@angstwad this appears to break one of the unit tests:

```
======================================================================
FAIL: test_callback (TestCallback.TestInventory)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jtanner/ansible/test/TestCallback.py", line 52, in test_callback
    assert os.path.exists(FILE_RUN)
AssertionError:
    self.run_ansible_playbook()
>>  assert <module 'os' from '/usr/lib/python2.7/os.pyc'>.path.exists('/tmp/ansible_test_finish')
    self.clean_file()


----------------------------------------------------------------------
Ran 96 tests in 19.617s

FAILED (SKIP=1, failures=1)
make: *** [tests] Error 1
```

--- a/lib/ansible/playbook/play.py
+++ b/lib/ansible/playbook/play.py
@@ -125,6 +125,11 @@ class Play(object):
         self.su               = ds.get('su', self.playbook.su)
         self.su_user          = ds.get('su_user', self.playbook.su_user)
 
+        # Fail out if user specifies a sudo param with a su param in a given play
+        if (ds.get('sudo') or ds.get('sudo_user')) and (ds.get('su') or ds.get('su_user')):
+            raise errors.AnsibleError('sudo params ("sudo", "sudo_user") and su params '
+                                      '("su", "su_user") cannot be used together')
+
         load_vars = {}
         load_vars['playbook_dir'] = self.basedir
         if self.playbook.inventory.basedir() is not None:

--- a/lib/ansible/playbook/task.py
+++ b/lib/ansible/playbook/task.py
@@ -157,6 +157,13 @@ class Task(object):
             self.su_user      = ds.get('su_user', play.su_user)
             self.su_pass      = ds.get('su_pass', play.playbook.su_pass)
 
+        # Fail out if user specifies a sudo param with a su param in a given play
+        if (ds.get('sudo') or ds.get('sudo_user') or ds.get('sudo_pass')) and \
+                (ds.get('su') or ds.get('su_user') or ds.get('su_pass')):
+            raise errors.AnsibleError('sudo params ("sudo", "sudo_user", "sudo_pass") '
+                                      'and su params "su", "su_user", "su_pass") '
+                                      'cannot be used together')
+
         # Both are defined
         if ('action' in ds) and ('local_action' in ds):
             raise errors.AnsibleError("the 'action' and 'local_action' attributes can not be used together")

